pkg/config/config.go

```diff
@@ -140,12 +141,27 @@ func (c *Config) InitializeConfig(configuration io.ReadCloser, defaults []io.Rea
 
 	// consider custom defaults defined in config.yml unless told otherwise
 	if ignoreCustomDefaults {
+		c.CustomDefaults = []string{}
 		log.Entry().Info("Ignoring custom defaults from pipeline config")
-	} else if c.CustomDefaults != nil && len(c.CustomDefaults) > 0 {
+	}
+
+	// retrieve custom default configuration passed via commonPipelineEnvironment
+	// e.g. when pipeline has been initialized in any CI/CD system, e.g. Jenkins
+	// customDefaults take precedense over defaults from environment
+	customDefaultsJSON := piperenv.GetResourceParameter(envRootPath, "commonPipelineEnvironment", "customDefaults")
+	var customDefaultsFromEnv []string
+	json.Unmarshal([]byte(customDefaultsJSON), &customDefaultsFromEnv)
+	if len(customDefaultsFromEnv) > 0 {
+		log.Entry().Info("using custom defaults provided via environment")
+		c.CustomDefaults = append(customDefaultsFromEnv, c.CustomDefaults...)
```

Comment:
The comment says it should take precedence. But I wonder if the append call shouldn't be the other way around then. I thought config from earlier entries in customDefaults overwrite config from later entries.

Reply:
Duh... sorry. I should learn to read.